Build/libHttpClient.Android/src/main/java/com/xbox/httpclient/NetworkObserver.java

```diff
@@ -0,0 +1,145 @@
+package com.xbox.httpclient;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.LinkProperties;
+import android.net.Network;
+import android.net.NetworkCapabilities;
+import android.net.NetworkRequest;
+import android.os.Build;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class NetworkObserver {
+    @NotNull private static String s_lastCapabilities = "";
+    @NotNull private static String s_lastLinkProperties = "";
+
+    @SuppressWarnings("unused")
+    public static void Initialize(Context appContext) {
+        ConnectivityManager cm = (ConnectivityManager) appContext.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        // Build a network request to query for all networks that "should" be
```

Comment:
Does this mean you're querying all available WiFi/cellular networks for the device and seeing which ones provide internet access?

Reply:
Yup, more or less. We construct a `NetworkRequest` that has some requirements - the only one we're setting here is that the network theoretically be able to access the internet (the system won't check that the network *can* actually reach the internet, but will know if a network *should* be able to) - and then that acts essentially as a kind of filter/query that selects the networks we want to be notified about in our callback.